views/templates/admin/smaily_configure.tpl

```diff
@@ -181,6 +181,38 @@
               </p>
             </div>
           </div>
+          <div class="form-group">
+            <label class="control-label col-lg-2">
+              {l s="Trigger Opt-in If Customer Joins With Newsletter Subscription" mod='smailyforprestashop'}
+            </label>
+            <div class="col-lg-10">
+              <span class="switch prestashop-switch fixed-width-lg">
+                <input type="radio" name="SMAILY_ENABLE_OPTIN_IF_CUSTOMER_JOINS_WITH_SUBSCRIPTION" id="Enable" value="1" {($smaily_enable_optin_if_customer_joins_with_subscription == 1) ? 'checked' :'' }>
+                <label for="Enable">{l s="Yes" mod='smailyforprestashop'}</label>
+                <input type="radio" name="SMAILY_ENABLE_OPTIN_IF_CUSTOMER_JOINS_WITH_SUBSCRIPTION" id="Disable" value="0" {($smaily_enable_optin_if_customer_joins_with_subscription == 0) ? 'checked' :'' }>
+                <label for="Disable">{l s="No" mod='smailyforprestashop'}</label>
+                <a class="slide-button btn"></a>
+              </span>
+            </div>
+          </div>
+          <div class="form-group">
+            <label class="control-label col-lg-2">
+              {l s="Opt-in Automation If Customer Joins With Newsletter Subscription" mod='smailyforprestashop'}
+            </label>
+            <div class="col-lg-10">
+              <select name="SMAILY_CUSTOMER_JOIN_AUTORESPONDER" id="SMAILY_CUSTOMER_JOIN_AUTORESPONDER">
+                {if empty($smaily_customer_join_autoresponder)}
+                <option value="">
+                  {l s="No Autoresponder" mod='smailyforprestashop'}
+                </option>
+                {else}
+                <option value='{$smaily_customer_join_autoresponder|json_encode}'>
```

Comment:
Since autoresponders are fetched only one time, when credentials are validated, how should we display the autoresponder title?

Reply:
You wouldn't need to display the autoresponder title separately. You could just add the `selected` attribute to option with value of `SMAILY_OPTIN_AUTORESPONDER`.

You could achieve this by reworking how `ajaxProcessGetAutoresponders` method works.

Instead of returning autoresponders as:
```json
{
  "success": true,
  "autoresponders": [
    {"id":14,"trigger_type":"form_submitted","title":"GOTO","is_enabled":false},
    {"id":15,"trigger_type":"form_submitted","title":"GOTO without loop","is_enabled":false}
  ]
}
```

You could do:
```json
{
  "success": true,
  "autoresponders": [
    {"id":14,"title":"GOTO","is_selected":false},
    {"id":15,"title":"GOTO without loop","is_selected":true}
  ]
}
```

And when you need to define by which configuration option a autoresponder should be selected by, you could provide a data field (or a query argument) in the query.

Though this is kind of a hackish approach.

An alternative (and a better solution) would be to use Javascript in controlling which option gets selected. Although that would mean that you would need to provide some sort of settings object to Javascript.

A simplest way to achieve that is to make a separate AJAX request to fetch some of the settings and use these values when rendering autoresponder options. An important thing to note here is that options rendering should only happen after first request has succeeded, as AJAX requests are always asynchronous you cannot guarantee that settings have been loaded before autoresponder options rendering has begun.

If you have any questions regarding the proposed solutions or there is something unclear, don't hesitate to ask.

Reply:
Only saving autoresponder ID to database. Also noticed user doesn't have a choice to select no autoresponder, added a default empty option to optin autoresponder list.
Used the same method for abandoned cart autoresponders aswell.